ntbs-service/wwwroot/source/Components/ValidateRelatedNotification.ts

```diff
@@ -0,0 +1,63 @@
+import Vue from "vue";
+import { getHeaders, getValidationPath } from "../helpers";
+import axios from "axios";
+
+const ValidateRelatedNotification = Vue.extend({
+    props: ["model"],
+    data: function() {
+        return {
+            relatedNotification: {},
+            isValid: false
+        }
+    },
+    mounted: function() {
+        var currentId = this.$refs["inputField"].value;
+        if (currentId) {
+            this.tryGetNotification(currentId);
+        }
+    },
+    methods: {
+        validate: function (event: FocusEvent) {
+            const inputField = event.target as HTMLInputElement;
+            const newValue = inputField.value;
+
+            this.tryGetNotification(newValue);
+        },
+        tryGetNotification: function (id: string) {
+            const requestConfig = {
+                url: `${getValidationPath(this.$props.model)}RelatedNotification`,
+                headers: getHeaders(),
+                params: {
+                    "value": id,
+                }
+            };
+            axios.request(requestConfig)
+                .then((response: any) => {
+                    var validationMessage = response.data;
+                    if (typeof validationMessage == 'string' && validationMessage) {
+                        this.$el.classList.add("nhsuk-form-group--error");
+                        this.$refs["errorField"].classList.remove("hidden");
+                        this.$refs["inputField"].classList.add("nhsuk-input--error");
+                        this.$refs["errorField"].textContent = validationMessage;
+                        this.relatedNotification = {};
+                        this.isValid = false;
+                    } else {
+                        // either blank or valid
+                        this.$el.classList.remove("nhsuk-form-group--error");
+                        this.$refs["errorField"].classList.add("hidden");
+                        this.$refs["inputField"].classList.remove("nhsuk-input--error");
+                        this.$refs["errorField"].textContent = '';
+                        this.relatedNotification = validationMessage ? validationMessage : {};

```

Comment:
name of variable validationMessage here is slightly deceiving, could we raname this to a more generic `response` or even return an object of `{validationMessage: X, relatedNotification: {}}` from the server

Reply:
I agree I'm kind of combining two concepts here - I like your suggestion of returning a an object that contains the message and the notification separately, that'll be much cleaner!